controllers/operator_controller.go

```diff
@@ -257,10 +257,16 @@ func (r *OperatorReconciler) ensureBundleDeployment(ctx context.Context, desired
 	}
 
 	// If the existing BD already has everything that the desired BD has, no need to contact the API server.
+	// Make sure the status of the existingBD from the server is as expected.
 	if equality.Semantic.DeepDerivative(desiredBundleDeployment, existingBundleDeployment) {
-		return nil
+		return verifySuccessfulBDStatus(existingBundleDeployment)
```

Comment:
I was thinking more along these lines:
```go
*desiredBundleDeployment = *existingBundleDeployment
return nil
```

And then have a separate function that maps a BD status to a Operator ready condition. So we wouldn't be verifying success so much as we'd be saying: "this BD status results in this Operator Ready condition"

WDYT?